http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -545,6 +546,16 @@ public void close() {
 
             @Override
             public <I, O, E> io.micronaut.http.HttpResponse<O> exchange(io.micronaut.http.HttpRequest<I> request, Argument<O> bodyType, Argument<E> errorType) {
+                if (!configuration.isAllowBlockEventLoop() && Thread.currentThread() instanceof FastThreadLocalThread) {
+                    throw new HttpClientException("""
+                        You are trying to run a BlockingHttpClient operation on a netty event \
+                        loop thread. This is a common cause for bugs: Event loops should \
+                        never be blocked. You can either mark your controller as \
+                        @ExecuteOn(TaskExecutors.BLOCKING), or use the reactive HTTP client \
+                        to resolve this bug. There is also a configuration option to \
```

Comment:
```suggestion
                        to resolve this bug. There is also a configuration option micronaut.http.client.allow-block-event-loop=true to \
```

>  It also mentions that there's a config property for this, but doesn't explain which, because I want to discourage an "easy fix" that will leave the bug in place.

I don't agree with this. I think we should say which is the config property.

Reply:
people will just enable the setting and move on without fixing their bug

Reply:
I understand you are trying to encourage correct behavior. 

However, we should not have secret properties they have to seek. If a setting exists, we have to document it. Let's say it is discouraged, and we can warn users about user it. But, we cannot expect our users to clone micronaut core and dive into our codebase to understand what is available.

Reply:
Why not linking to a place in the docs where we explain why this is bad, and also document the property in a warning box?